src/Microsoft.Diagnostics.Monitoring.HostingStartup/ParameterCapturing/MethodResolver.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Diagnostics.Monitoring.StartupHook.MonitorMessageDispatcher.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace Microsoft.Diagnostics.Monitoring.HostingStartup.ParameterCapturing
+{
+    public sealed class MethodResolver
+    {
+        private sealed class DeclaringTypeDescription : IEquatable<DeclaringTypeDescription?>
```

Comment:
Using a [record class](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/record) would eliminate the need to write this out.